lib/DataHarmonizer.js

```diff
@@ -2182,6 +2168,44 @@ class DataHarmonizer {
       } // column/field loop end
     } // row loop end
 
+    // Check row uniqueness for identifier fields and unique_key sets
+    const doUniqueValidation = (columnNumbers) => {
+      const values = columnNumbers.map((columnNumber) => {
+        if (columnNumber < 0) {
+          return [];
+        } else {
+          return this.hot
+            .getDataAtCol(columnNumber)
+            .filter((val) => val != null);
```

Comment:
Are nulls allowed in unique_keys?  If not, error report for cell's that have them?  
If nulls allowed, then what about scenario:
             keycol1    keycol2 other fields
row 1    null           A           yada
row 2   A               null       yadada
row 3   A               B           foobar

The above would satisfy unique keys where nulls are allowed?

Reply:
I believe nulls should be allowed (assuming the column itself is not required). The purpose of the `filter` call in the above section was intended to eliminate _completely_ empty rows. So I need to fix that.

Reply:
The latest revision does more bookkeeping about empty rows in order to not take them into account when doing uniqueness checks.